nil/core-concepts/contract-co-location.mdx

```diff
@@ -0,0 +1,118 @@
+# Contract co-location
+
+## Definition
+
+[**Cross-shard communications**](./shards-parallel-execution#message-passing-checks) are a built-in feature of the =nil; protocol ensuring horizontal scalability with no fragmentation.
+
+However, =nil; also supports contract co-location, which is a technique for ensuring that two accounts are located on the same shard. In this case, interaction between these accounts does not require cross-shard communications.
+
+In plain terms, co-location is a request made to change the value stored under the account's address in [**the mapping of accounts and shards**](./accounts) stored in the consensus shard. 
+
+:::note[Explanation]
+
+When two accounts (Account 1 and Account 2) are co-located, communications between them can occur synchronously similarly to regular Ethereum transactions. 
+
+However, this is not the case if Account 1 and Account 2 are placed in different shards and Account 1 calls a function in Account 2. In this situation, the transaction from Account 1 must create another transaction that subsequently performs an async call to the function in Account 2. This async transaction could then be executed in a future block in the shard where Account 2 is located.
+
+This ensures that sharding does not result in fragmentation.
+
+:::
+
+## Scalability considerations
+
+On the one hand, contract co-location presents a valuable opportunity for developers wanting to ensure the lowest possible transaction processing times for their users.
+
+On the other hand, co-location encourages having as many accounts as possible on one shard, undermining the benefits of horizontal scaling.
+
+To mitigate this, =nil; imposes economic and ownership restrictions on account co-location. These restrictions are expressed in terms of co-location depth, which is the number of addresses ($N$) a given address is co-located with. 
```

Comment:
Economic formulae is outdated and related to the previous co-location model, ownership restriction is not used anymore

Reply:
Got it, formula has been removed